File: pset6/credit/credit.py
from cs50 import get_int

while True:
    number = get_int("enter credit card number: ")
    if number > 0:
        break

j = 0
a = []
while number != 0:
    digits = number % 10
    number = number - digits
    number = number / 10
    a.append(int(digits))
    j += 1
p = j
# for i in range(len(a)):
#   print(a[i],end="")
k = 0
es = 0
es1 = 0
os = 0
for i in range(int(p / 2)):
    if a[2 * i + 1] > 4:
        l = 2 * a[2 * i + 1]
        while l > 0:
            x = l % 10
            k = k + x
            l = l - x
            l = l / 10

        es = es + k
        k = 0
    else:
        es1 = 2 * a[2 * i + 1] + es1

if p % 2 != 0:
    for z in range(int((p / 2) + 1)):
        os = os + a[2 * z]
else:
    for y in range(int(p / 2)):
        os = os + a[2 * y]
counter = 0
if (os + es + es1) % 10 == 0:
    if p == 15 and (a[14] == 3 and (a[13] == 4 or a[13] == 7)):
        print("AMEX")
        counter = 1
    if p == 16 and (a[15] == 5 and (a[13] <= 5)):
        print("MASTERCARD")
        counter = 1
    if (p == 13 and a[12] == 4) or (p == 16 and a[15] == 4):
        print("VISA")
        counter = 1
    if counter == 0:
        print("INVALID")
else:
    print("INVALID")

File: pset1/credit/credit.c
#include<stdio.h>
#include<cs50.h>
#include<string.h>
int main(void)
{
    int counter = 0, digits, a[50], j = 0, k = 0, l, es = 0, os = 0, n, p, es1 = 0, x = 0, z;
    long int number;
    do
    {
        number = get_long("enter credit card number: ");
    }
    while (number < 0);

    while (number != 0)
    {
        digits = number % 10; //exracting last digit
        number = number / 10; //removing last digit
        a[j] = digits;
        j++;

    }
    p = j;



    for (n = 0; n < p / 2; n++) //check sum
    {
        if (a[2 * n + 1] > 4) // two digit or more individual digit sum
        {
            l = 2 * a[2 * n + 1];
            while (l != 0)
            {
                x = l % 10;
                k = k + x;
                l = l / 10;
            }
            es = es + k;
            k = 0;
        }


        else // even place sum
        {
            es1 = 2 * a[2 * n + 1] + es1;
        }
    }

    if (p % 2 != 0) //odd digit sum in odd place
    {
        for (z = 0; z <= p / 2; z++)
        {
            os = os + a[2 * z];
        }
    }
    else //even digit sum in odd place
        for (z = 0; z < p / 2; z++)
        {
            os = os + a[2 * z];
        }
    if ((os + es + es1) % 10 == 0) //check sum
    {
        if (p == 15 && (a[14] == 3 && (a[13] == 4 || a[13] == 7))) //americn express verification
        {
            printf("AMEX\n");
            counter = 1;
        }

        if (p == 16 && (a[15] == 5 && (a[13] <= 5))) //mastercard verification
        {
            printf("MASTERCARD\n");
            counter = 1;
        }
        if ((p == 13 && a[12] == 4) || (p == 16 && a[15] == 4)) //visa verification
        {
            printf("VISA\n");
            counter = 1;
        }
        if (counter == 0)
        {
            printf("INVALID\n");
        }

    }
    else
    {
        printf("INVALID\n");
    }
}